app/src/testDemo/java/com/google/samples/apps/nowinandroid/ui/NiaAppScreenSizesScreenshotTests.kt

```diff
@@ -0,0 +1,233 @@
+import com.google.samples.apps.nowinandroid.core.testing.data.topicsTestData
+import com.google.samples.apps.nowinandroid.core.testing.util.DefaultRoborazziOptions
+import com.google.samples.apps.nowinandroid.uitesthiltmanifest.HiltComponentActivity
+import dagger.hilt.android.testing.BindValue
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.HiltTestApplication
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.GraphicsMode
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+
+/**
+ * Tests that the navigation UI is rendered correctly on different screen sizes.
+ */
+@OptIn(ExperimentalMaterial3WindowSizeClassApi::class)
+@RunWith(RobolectricTestRunner::class)
+@GraphicsMode(GraphicsMode.Mode.NATIVE)
+// Configure Robolectric to use a very large screen size that can fit all of the test sizes.
+// This allows enough room to render the content under test without clipping or scaling.
+@Config(application = HiltTestApplication::class, qualifiers = "w1000dp-h1000dp", sdk = [33])
+@LooperMode(LooperMode.Mode.PAUSED)
+@HiltAndroidTest
+class NiaAppScreenSizesScreenshotTests {
+
+    /**
+     * Manages the components' state and is used to perform injection on your test
+     */
+    @get:Rule(order = 0)
+    val hiltRule = HiltAndroidRule(this)
+
+    /**
+     * Create a temporary folder used to create a Data Store file. This guarantees that
+     * the file is removed in between each test, preventing a crash.
+     */
+    @BindValue
+    @get:Rule(order = 1)
+    val tmpFolder: TemporaryFolder = TemporaryFolder.builder().assureDeletion().build()
+
+    /**
+     * Use a test activity to set the content on.
+     */
+    @get:Rule(order = 2)
+    val composeTestRule = createAndroidComposeRule<HiltComponentActivity>()
+
+    @Inject
+    lateinit var networkMonitor: NetworkMonitor
+
+    @Inject
+    lateinit var userDataRepository: UserDataRepository
+
+    @Inject
+    lateinit var newsRepository: NewsRepository
+
+    @Before
+    fun setup() {
+        val config = Configuration.Builder()
+            .setMinimumLoggingLevel(Log.DEBUG)
+            .setExecutor(SynchronousExecutor())
+            .build()
+
+        // Initialize WorkManager for instrumentation tests.
+        WorkManagerTestInitHelper.initializeTestWorkManager(
+            InstrumentationRegistry.getInstrumentation().context,
+            config,
+        )
+
+        hiltRule.inject()
+
+        // Configure user data
+        runBlocking {
+            userDataRepository.setShouldHideOnboarding(true)
+            userDataRepository.setFollowedTopicIds(setOf(topicsTestData.first().id))
```

Comment:
Since we are using Hilt, shouldn't we use an injected `TopicsRepository` instead of accessing fake test data?

Reply:
Good catch! I was being lazy here. Now I'm accessing the topic ID correctly.